config-reloader/template/template.go

```diff
@@ -0,0 +1,144 @@
+package template
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	konfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/yaml"
+)
+
+var k8sClient client.Reader
+
+// Render is a go template rendering function it includes all the sprig lib functions
+// as well as some extras like a k8sLookup function to get values from k8s objects
+// you can access environment variables from the template under .Env
+// The passed values will be available under .Values in the templates
+func Render(out io.Writer, tmpl string, values interface{}) error {
```

Comment:
The Render function should also handle errors returned by t.Execute. It should return any errors encountered during template execution.

Reply:
You mean wrapping the error into some other error? It does already return the errors from template execution, what I could do here is to also return the parsing errors instead of using `Must`...

Reply:
I've updated the code to return the errors from `Parse` instead of using `Must`